packages/ui/src/primitives/tailwind/Radio/index.tsx

```diff
@@ -23,56 +23,78 @@ All portions of the code written by the Infinite Reality Engine team are Copyrig
-  const radioId = useId()
+const innerCircleSizeVariant = {
+  sm: 'h-2 w-2',
+  md: 'h-2.5 w-2.5'
+}
+
+export const Radio = ({ disabled, label, onClick, value, description, checked, variant = 'sm' }: RadioProps) => {
+  const handleClick = (event: React.MouseEvent) => {
+    event.stopPropagation()
+    if (disabled) return
+    onClick?.(value)
+  }
+  const handleKeyUp = (event: React.KeyboardEvent) => {
+    if (disabled) return
+    if (['Enter', ' '].includes(event.key)) {
+      onClick?.(value)
+    }
+  }
+
   return (
-    <div className="flex flex-col">
-      <div className="flex items-start gap-2">
-        <div className="mt-1 grid place-items-center">
-          <input
-            id={radioId}
-            type="radio"
-            name={name}
-            disabled={disabled}
-            className="
-            focus:ring-primary-blue
-            peer col-start-1
-            row-start-1 h-4
-            w-4 shrink-0 appearance-none rounded-full border-[1.5px]
-            border-blue-500 focus:outline-none focus:ring-[0.5px] focus:ring-offset-0
-            disabled:border-gray-400
-          "
-            onChange={onChange}
-            defaultChecked={defaultChecked}
-            value={value}
-          />
-          <div
-            className={twMerge(
-              'pointer-events-none',
-              'col-start-1 row-start-1',
-              'h-2 w-2 rounded-full peer-checked:bg-blue-500',
-              'peer-checked:peer-disabled:bg-gray-400'
-            )}
-          />
-        </div>
-        <label htmlFor={radioId} className={twMerge('text-start hover:cursor-pointer', disabled && 'text-gray-400')}>
-          {label}
-        </label>
+    <div
+      className="group flex items-center gap-x-2 outline-none"
+      tabIndex={1}
+      onClick={handleClick}
+      onKeyUp={handleKeyUp}
+    >
+      <div
+        className={twMerge(
+          outerCircleVariant[variant],
+          'grid cursor-pointer place-items-center rounded-full border',
+          !disabled && 'border-[#212226] group-hover:border-[#9CA0AA] group-focus:border-[#375DAF]',
+          !disabled && 'bg-[#141619] group-hover:bg-[#191B1F] group-focus:bg-[#212226]',
+          disabled && 'cursor-not-allowed border-[#42454D] bg-[#191B1F]'
+        )}
+      >
+        <div
+          className={twMerge(
+            innerCircleSizeVariant[variant],
+            'block rounded-full',
+            !checked && 'hidden',
+            !disabled && 'bg-[#5F7DBF] group-hover:bg-[#5F7DBF] group-focus:bg-[#5F7DBF]',
+            disabled && 'bg-[#42454D]'
+          )}
+        />
+      </div>
+      <div
+        className={twMerge(
+          'flex flex-col',
+          !disabled && 'text-start text-[#D3D5D9] group-hover:cursor-pointer',
+          disabled && 'text-[#6B6F78]'
+        )}
+      >
+        <span className="group-hover:text-[#F5F5F5] group-focus:text-[#F5F5F5]">{label}</span>
+        <span>{description}</span>
```

Comment:
![image](https://github.com/user-attachments/assets/31d387fd-70dc-42bc-8425-fcaadc326208)

When Radio is disabled, hover styles should be disabled for description too.